File: frost-scripts/performance-test/match-expression-workload.py
#!/usr/bin/env python3

n = 300_000

values = []
for i in range(n):
    mod = i % 4
    if mod == 0:
        values.append(i)
    elif mod == 1:
        values.append(float(i))
    elif mod == 2:
        values.append(str(i))
    else:
        values.append(i % 2 == 0)


def classify(v):
    if isinstance(v, bool):
        return 1
    elif isinstance(v, int):
        return v
    elif isinstance(v, float):
        return 2
    elif isinstance(v, str):
        return len(v)
    return 0


result = sum(classify(v) for v in values)
assert result == 11_250_497_223
print(result)

File: frost-scripts/performance-test/match-expression-workload.rb
#!/usr/bin/env ruby

n = 300_000

values = (0...n).map do |i|
  case i % 4
  when 0 then i
  when 1 then i.to_f
  when 2 then i.to_s
  else i.even?
  end
end

def classify(v)
  case v
  when true, false then 1
  when Integer then v
  when Float then 2
  when String then v.length
  else 0
  end
end

result = values.sum { |v| classify(v) }
raise unless result == 11_250_497_223
puts result
